src/ConnectionAutomaton.cpp

```diff
@@ -98,9 +101,11 @@ void ConnectionAutomaton::closeDuplexConnection(folly::exception_wrapper ex) {
 }
 
 void ConnectionAutomaton::reconnect(
-    std::unique_ptr<DuplexConnection> newConnection) {
+  std::unique_ptr<DuplexConnection> newConnection,
+  std::unique_ptr<ClientResumeStatusCallback> statusCallback) {
   disconnect();
   connection_ = std::shared_ptr<DuplexConnection>(std::move(newConnection));
+  resumeStatusCallback_ = std::shared_ptr<ClientResumeStatusCallback>(std::move(statusCallback));
```

Comment:
you probably don't want to allow more than one resume operation if flight, so I would say you want to CHECK(!resumeStatusCallback)